src/overlays/actors/ovl_En_Thiefbird/z_en_thiefbird.c

```diff
@@ -93,93 +146,1020 @@ static DamageTable D_80C13644 = {
+            if (this->collider.elements[i].info.bumperFlags & 2) {
+                break;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/EnThiefbird_Destroy.s")
+        if (this->actor.colChkInfo.damageEffect == 3) {
+            func_80C11454(this);
+        } else if (this->actor.colChkInfo.damageEffect == 4) {
+            this->unk_18C = 20;
+            this->unk_3D8 = 0.5f;
+            this->unk_3D4 = 4.0f;
+            if (i != ARRAY_COUNT(this->colliderElements)) {
+                sph = &this->collider.elements[i];
+                Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, sph->info.bumper.hitPos.x,
+                            sph->info.bumper.hitPos.y, sph->info.bumper.hitPos.z, 0, 0, 0, 4);
+            }
+        } else if (this->actor.colChkInfo.damageEffect == 2) {
+            this->unk_18C = 0;
+            this->unk_3D8 = 0.5f;
+            this->unk_3D4 = 4.0f;
+        } else if (this->actor.colChkInfo.damageEffect == 5) {
+            this->unk_18C = 0x1E;
+            this->unk_3D8 = 0.5f;
+            this->unk_3D4 = 2.0f;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C10984.s")
+        if (this->unk_3E8 != 0) {
+            this->actor.colChkInfo.damage = 0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C10B0C.s")
+        if (!Actor_ApplyDamage(&this->actor)) {
+            this->actor.colChkInfo.health = 0;
+            Enemy_StartFinishingBlow(globalCtx, &this->actor);
+            func_80C10984(this, 40);
+            func_80C11C60(this);
+        } else {
+            func_80C11F6C(this, globalCtx);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C10DE8.s")
+void func_80C12D00(EnThiefbird* this) {
+    static Vec3f D_80C1368C = { 0.0f, -0.15f, 0.0f };
+    s32 i;
+    f32 phi_f20;
+    EnThiefbirdUnkStruct* ptr = &this->unk_3F0[0];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C10E98.s")
+    for (i = 0; i < ARRAY_COUNT(this->unk_3F0); i++, ptr++) {
+        if (ptr->unk_22) {
+            ptr->unk_22--;
+            Math_Vec3f_Sum(&ptr->unk_00, &ptr->unk_0C, &ptr->unk_00);
+            Math_Vec3f_Sum(&ptr->unk_0C, &D_80C1368C, &ptr->unk_0C);
+            ptr->unk_1C++;
+            Math_StepToF(&ptr->unk_0C.x, 0.0f, 0.05f);
+            Math_StepToF(&ptr->unk_0C.z, 0.0f, 0.05f);
+            if (ptr->unk_0C.y < -0.5f) {
+                ptr->unk_0C.y = -0.5f;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11338.s")
+            if (i & 1) {
+                phi_f20 = -1.0f;
+            } else {
+                phi_f20 = 1.0f;
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Thiefbird/func_80C11454.s")
+            ptr->unk_20 = Math_SinS(ptr->unk_1C * 0x7D0) * 8192.0f;
```

Comment:
```suggestion
            ptr->unk_20 = Math_SinS(ptr->unk_1C * 0x7D0) * 0x2000
```
May need `(f32)0x2000`